src/MSBuild/XMake.cs

```diff
@@ -1292,6 +1303,25 @@ string outputResultsCache
             return success;
         }
 
+        private static bool PrintTargets(string projectFile, string toolsVersion, Dictionary<string, string> globalProperties, TextWriter targetsWriter, ProjectCollection projectCollection)
+        {
+            try
+            {
+                Project project = projectCollection.LoadProject(projectFile, globalProperties, toolsVersion);
+
+                targetsWriter.WriteLine(string.Join(Environment.NewLine, project.Targets.Keys));
+
+                projectCollection.UnloadProject(project);
+                return true;
+            }
+            catch (Exception ex)
+            {
+                var message = ResourceUtilities.FormatResourceStringStripCodeAndKeyword("TargetsCouldNotBePrinted", ex.Message);
```

Comment:
```suggestion
                var message = ResourceUtilities.FormatResourceStringStripCodeAndKeyword("TargetsCouldNotBePrinted", ex);
```

This preserves the call stack which can make fixing bugs that report this error MUCH easier.